cloudbot/util/decorators.py

```diff
@@ -0,0 +1,25 @@
+from time import time
+
+SPAM_LIST = {}
+
+
+def spam_guard(func):
+    def wrapper(*args, **kwargs):
+        nick = kwargs.get('nick', None)  # pull the nick from the args
+        if nick:  # if there is no nick specified for whatever reason, carry on
+            global SPAM_LIST
+            if SPAM_LIST.get(nick, None):
+                SPAM_LIST[nick][0] = time()
+                if time() - SPAM_LIST[nick][0] < SPAM_LIST[nick][1]:
+                    # notice that they are spamming
+                    SPAM_LIST[nick][1] += 10
+                    return (
+                        "{} attempted to execute a command within their previous" \
+                        " limit. An additional 10 seconds has been added. Current" \
+                        " limit is {} seconds.".format(nick, SPAM_LIST[nick][1])
```

Comment:
I'm not seeing any code to reset the limit back to 10 seconds -- so if a user spams within 10 seconds, their limit is extended to 20 permanently (or at least for the rest of the bot's runtime), and so on. Couldn't this eventually hit a point where the user can't use commands for some unreasonable time period, and eventually probably not at all?